torch/csrc/jit/tensorexpr/ir_simplifier.cpp

```diff
@@ -1871,6 +1871,61 @@ Stmt* TermExpander::mutate(const Free* v) {
   return new Free(buffer_var_new);
 }
 
+Stmt* TermExpander::fuseSyncThreads(Block* block) {
```

Comment:
Minor: this is fine. But it feels that this simplification fits better in the cuda_codegen, instead of part of a generic simplification. But it's up to you.

Reply:
If SyncThreads is really just a Cuda specific node then I guess that makes sense. I can move it to a new mutator.

Reply:
Actually this doesn't work nearly as well outside of the simplifier, I think unless you have strong objections I'll keep it where it is.